tests/color_convert.rs

```diff
@@ -0,0 +1,36 @@
+#[macro_use]
+extern crate approx;
+extern crate rustc_serialize;
+extern crate csv;
+extern crate palette;
+
+
+macro_rules! assert_color_eq {
+    ($a:expr, $b:expr, [$($components:ident),+]) => ({
+        $(
+            let a: f32 = $a.$components.into();
+            let b: f32 = $b.$components.into();
+            assert_relative_eq!(a, b, epsilon = 0.05);
+        )+
+    })
+}
+
+// Check if the hue diff equal to zero
+macro_rules! assert_color_hue_eq {
+    ($a:expr, $b:expr, [$($components:ident),+]) => ({
+        $(
+            let a: f32 = $a.$components.into();
+            let b: f32 = $b.$components.into();
+
+            let smallest_angle_diff = |ang1: f32, ang2: f32| -> f32 {
+                let a = ang1 - ang2;
+                (a + 180.0) % 360.0 - 180.0
+            };
+
+            let out = smallest_angle_diff(a, b);
```

Comment:
($a.$components - $b.$components).into() should have the same effect, assuming that they are hues.